staker/rollup_watcher.go

```diff
@@ -165,6 +165,44 @@ func (r *RollupWatcher) LookupNode(ctx context.Context, number uint64) (*NodeInf
 	}, nil
 }
 
+func (r *RollupWatcher) LookupNodeByBlockNumber(ctx context.Context, blockNumber uint64) (*NodeInfo, error) {
+	var query = ethereum.FilterQuery{
+		FromBlock: big.NewInt(int64(blockNumber)),
+		ToBlock:   big.NewInt(int64(blockNumber)),
+		Addresses: []common.Address{r.address},
+		Topics:    [][]common.Hash{{nodeCreatedID}},
+	}
+	logs, err := r.client.FilterLogs(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	if len(logs) == 0 {
```

Comment:
BTW - this function cannot be used for binary search. Binary search cannot be based on logs but on function calls.
Say in block 1000 we got message 7, and in block 1200 we got message 8 - checking block 1100 should say message 7 and not no message found, so we could do a binary search.
For this we should use things like delayedMessageCount and sequencerMessageCount in the bridge interface..